demo/paging/row-grouping.component.ts

```diff
@@ -0,0 +1,253 @@
+                group[index].exppayyes=0;
+                group[index].exppaypending=0;
+                group[index].exppayno=1;
+              }
+            }
+
+            if (group[index].exppayyes === 0 && group[index].exppayno === 0 && group[index].exppaypending === 0)
+              expectedPaymentDealtWith = false;
+            console.log('expectedPaymentDealtWith', expectedPaymentDealtWith);
+          }
+        }
+      }
+    }
+    else{
+      for (var index = 0; index < group.length; index++) {
+        if (group[index].exppayyes === 0 && group[index].exppayno === 0 && group[index].exppaypending === 0)
+          expectedPaymentDealtWith = false;
+        console.log('expectedPaymentDealtWith', expectedPaymentDealtWith);
+      }      
+    }
+
+    //check if there is a pending selected payment or a row that does not have any expected payment selected
+    if (group.filter(row => row.exppaypending===1).length===0 && group.filter(row => row.exppaypending===0 && row.exppayyes===0 && row.exppayno===0).length===0)
+    //if (expectedPaymentDealtWith)
+    {
+      console.log('expected payment dealt with')
+      //check if can set the group status
+      const numberOfExpPayYes = group.filter(row => row.exppayyes===1).length;
+      const numberOfSourceFunder = group.filter(row => row.exppayyes===1 && row.source==='Funder').length;
+      const numberOfSourceCalculated = group.filter(row => row.exppayyes===1 && row.source==='Calculated').length;
+      const numberOfSourceManual = group.filter(row => row.exppayyes===1 && row.source==='Manual').length;
+
+      console.log('numberOfExpPayYes', numberOfExpPayYes)
+      console.log('numberOfSourceFunder', numberOfSourceFunder)
+      console.log('numberOfSourceCalculated', numberOfSourceCalculated)
+      console.log('numberOfSourceManual', numberOfSourceManual)
+
+      if (numberOfExpPayYes>0)
+        if (numberOfExpPayYes === numberOfSourceFunder)
+          groupStatus = 'Funder Selected'
+        else if (numberOfExpPayYes === numberOfSourceCalculated)
+          groupStatus = 'Calculated Selected'
+        else if (numberOfExpPayYes === numberOfSourceManual)
+          groupStatus = 'Manual Selected'  
+        else
+          groupStatus = 'Hybrid Selected'
+    }
+
+    group[0].groupstatus = groupStatus;
+
+    console.log('group', group);
+
+  console.log('group.length', group.length)
+
+  console.log('event.target.value', event.target.value)
+
+    /*
+    for (var index = 0; index < group.length; index++) {
+      var element = group[index];
+      
+    }
+    console.log('group.length', group.length)
+    
+    
+    console.log('event.target.name', event.target.name)
+    console.log('event.target.id', event.target.id)
+    console.log('event.target.checked', event.target.checked)
+    */
+    
+  }
+
+  updateValue(event, cell, cellValue, row) {
+    this.editing[row.$$index + '-' + cell] = false;
```

Comment:
As part of removing references to $$index I reviewed the internal rowIndexes map inside body.component.ts and if the purposes of this map is to register the index of a row couldn't we use the indexOf the internal rows array? I am performing this change already but I thought better I ask before continuing.